app/github-login-button.tsx

```diff
@@ -0,0 +1,57 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import useSWR from 'swr';
+
+const fetcher = (url: string) => fetch(url).then((res) => res.json());
+
+const GitHubLoginButton: React.FC = () => {
+  const [token, setToken] = useState<string | null>();
+  const { data, error, isLoading } = useSWR(token ? `/api/github/user/${token}` : null, fetcher);
+
+  useEffect(() => {
+    setToken(localStorage.getItem('github_access_token'));
```

Comment:
If you're looking for something that works server-side, too, maybe [`cookies()`](https://nextjs.org/docs/app/api-reference/functions/cookies)?

Also, I'm really surprised that `useEffect` is necessary to delay this localStorage call. I'd think `use client` would be enough.